transaction/interfaces.py

```diff
@@ -22,12 +22,19 @@ class ITransactionManager(Interface):
     """
 
     def begin():
-        """Begin a new transaction.
+        """Explicitly begin a new transaction.
 
         If an existing transaction is in progress, it will be aborted.
 
-        The newTransaction() method of registered synchronizers is called,
+        The ``newTransaction`` method of registered synchronizers is called,
         passing the new transaction object.
+
+        Note that transactions may be started implicitly without
+        calling ``begin``. In that case, ``newTransaction`` isn't
+        called because the transaction manager doesn't know when to
+        call it.  The transaction is likely to have begin long before
```

Comment:
Actually, *is* there a way other than using ``get`` to create a transaction? It seems to me that the transaction manager is always involved in starting transactions. But what it may not be involved in is work done outside the actual transaction boundaries. 

Perhaps there's a terminology problem here. There's an actual Transaction object, and the transaction manager is always aware of when that begins (because it creates it). But there are people that do work before a Transaction has begun, and would like that work to be reflected in an implicitly begun Transaction; they think of all the work they do as a logical transaction (little t). This would be fine, except there may be synchronizers that do "destructive" things when a Transaction is begun (ZODB polling being a prominent example).

The short answer is "you're doing it wrong" if you expect that to work, IMHO. This is a logical inconsistency in the API and the way that Transactions work. 

Can we deprecate the implicit-begin behaviour of get(), such that sometime in the future it either returns None if no transaction has begun (which will make debugging the issue easier, callers will get exceptions immediately if they assume a transaction can just be started) or calls newTransaction if there is no transaction (potentially less invasive, but also potentially harder to debug)? This is probably the wrong place to discuss this...